src/interactions/music/add.ts

```diff
@@ -0,0 +1,76 @@
+import {
+    CommandInteraction,
+    GuildMember,
+    SlashCommandIntegerOption,
+    SlashCommandStringOption,
+    VoiceChannel,
+} from 'discord.js';
+import { search } from '../../services/search';
+import Interaction from '../../models/Interaction';
+
+export default class Add extends Interaction<CommandInteraction> {
+    name = 'add';
+
+    description = 'add something to the queue';
+
+    options = [
+        new SlashCommandStringOption()
+            .setName('query')
+            .setDescription('Enter keyword or youtube url.')
+            .setRequired(true),
+        new SlashCommandIntegerOption()
+            .setName('tracknumber')
+            .setDescription(
+                'Enter the position you want to insert the track at in queue',
+            )
+            .setRequired(false),
+    ];
+
+    execute = async (interaction: CommandInteraction) => {
+        const player = this.client.musicPlayers.getOrCreate(
+            interaction.guildId!,
```

Comment:
Why the `!` at the end? If it's to indicate this can't be undefined, I've seen standard practice to be TypeScript casting, though I admit `!` is shorter.

Reply:
[Non-null assertion operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator)